DMTN-148.tex

```diff
@@ -50,9 +50,255 @@
 % ADD CONTENT HERE
 % You can also use the \input command to include several content files.
 
+\section{Intro}
+
+The goal of this tech note is to provide a plan for the handling of
+calibration products in the Gen-3 Butler environment.  There are a
+large number of these products currently in use, but many of them do
+not have a clearly defined process to define, ingest, test, and
+deprecate them.  The lack of documentation on how these processes work
+has led to many implementations of similar paths, further complicating
+the issue.
+
+
+\section{The End State}
+
+The final state that we wish to get to has the following properties:
+
+\begin{itemize}
+\item All calibration products are either lsst.afw.image.Exposure
+  image data or an explicit calibration class in the \verb|ip_isr|
+  package.\footnote{This will require class reorganization to migrate
+    the Defects and Curve calibrations definitions.}
+\item All calibration products should be constructed from a DM Task in
+  the \verb|cp_pipe| package, and have the input data and
+  configuration parameters persisted with the calibration itself.
+  This guarantees that any product can be reconstructed in the future.
+  Exceptions will exist for external or lab created calibrations,
+  although the majority of used calibrations will likely be produced
+  directly.
+\item These provenance, config, and final product files may be saved
+  to an \verb|obs_CAMERA_data| package in git for convenience.  This also allows
+  these files to be migrated to other Butler repositories in a
+  consistent fashion.
```

Comment:
What's canonical? With the same data in potentially multiple Butler repositories and saved to git, how do we resolve discrepancies?

Reply:
They will have different collection names. I was considering deriving the collection name from the sha1 of the data package.

Reply:
The Butler repo that generates/validates/certifies the calibration explicitly accepts that calibration as canonical.  Calibrations that are ingested from the obs packages have valid ranges from some validation/certification process.
I have assumed that the number of Butler repos that process data from raw will be small, so haven't thought there would be discrepancies.